.github/workflows/tests.yml

```diff
@@ -0,0 +1,90 @@
+name: build and test
+
+on: [push, pull_request]
+
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        go: ['1.14', '1.15']
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          # clone in the gopath
+          path: src/github.com/${{ github.repository }}
+      - uses: actions/setup-go@v2
+        with:
+          stable: false
+          go-version: ${{ matrix.go }}
+      - name: setup environment
+        run: |
+          echo "GOPATH=$GITHUB_WORKSPACE" >> $GITHUB_ENV
+          echo "GO111MODULE=on" >> $GITHUB_ENV
+      - name: build coredhcp
+        run: |
+          set -exu
+          cd $GITHUB_WORKSPACE/src/github.com/${{ github.repository }}/cmds/coredhcp
+          go build
+  coredhcp-generator:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        go: ['1.14', '1.15']
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          # clone in the gopath
+          path: src/github.com/${{ github.repository }}
+      - uses: actions/setup-go@v2
+        with:
+          stable: false
+          go-version: ${{ matrix.go }}
+      - name: setup environment
+        run: |
+          echo "GOPATH=$GITHUB_WORKSPACE" >> $GITHUB_ENV
+          echo "GOBIN=$GITHUB_WORKSPACE/bin" >> $GITHUB_ENV
+          echo "GO111MODULE=off" >> $GITHUB_ENV # until we solve this for coredhcp-generator
```

Comment:
Any reason to use it like this instead of the `env:` attribute https://docs.github.com/en/actions/reference/environment-variables#about-environment-variables ? (worth adding in a comment if there's a specific reason to do it like this)

Reply:
Can't remember the details, but one reason is that `env` doesn't let you dynamically construct variables from other variables. I just reused this approach from another configuration that I wrote, which in this case would be only necessary for GOBIN. I can move the other two to `env`, and leave a comment for GOBIN

Reply:
actually only GO111MODULE=off can be moved to `env`, in which case I'd rather keep all the variables there for consistency

Reply:
I'll add a comment

Reply:
done